source/deploy/postgres-migration-assist-tool.rst

```diff
@@ -142,10 +124,47 @@ This is the final step of the migration process, where we need to update the Mat
 
   If your configuration was stored in the database, see the :ref:`configuration in database <deploy/manual-postgres-migration:configuration in database>` documentation for migration details. Once migrated, you should also update the ``MM_CONFIG`` environment variable to point to the new DSN.
 
+Air-gapped environments
+------------------------
+
+The ``migration-assist`` tool contains all necessary files for migration, removing any need for external access to repositories. Once transferred across the air gap, the tool operates independently. Follow the guidance below to perform a seamless MySQL to PostgreSQL migration in an air-gapped environment without needing external dependencies:
+
+1. Verify that the ``migration-assist`` binary is the latest version available to benefit from improvements and fixes.
+
+2. Transfer the latest version of the ``migration-assist`` binary into the air-gapped environment.
+
+3. Generate the schema and data output file from MySQL using the ``--fix`` flags for Unicode, artifacts, and VARCHAR. This step is crucial to ensure the MySQL database is ready for migration. The following command checks the MySQL database schema, applies necessary fixes, and automatically produces a file named ``mysql.output``:
+
+   .. code-block:: sh
+
+      migration-assist mysql "user:password@tcp(localhost:3306)/mattermost" --fix-unicode --fix-artifacts --fix-varchar
+
+4. Use the ``mysql.output`` file to apply the schema and data into the PostgreSQL database:
+
+   .. code-block:: sh
+
+      migration-assist postgres "postgres://mmuser:mostest@localhost:5432/imported?sslmode=disable" --run-migrations --applied-migrations="./mysql.output"
+
+5. Follow `step 3 <#step-3-generate-a-pgloader-configuration>`__ through `step 7 <#step-7-configure-mattermost-to-utilize-the-new-postgresql-database>`__ above to complete the migration process using the ``migration-assist`` tool.
+
 Compile the migration-assist tool
 ---------------------------------
 
-The ``migration-assist`` tool can be downloaded and compiled with the `Go <https://go.dev/>`__ toolchain. The tool requires at least ``v1.22`` of the Go compiler.
+The ``migration-assist`` tool offers 3 core utility commands:
+
+1. ``migration-assist mysql``
+
+   Checks the MySQL database schema to ensure readiness for migration, and offers fixes for common issues. See the `install <#install>`__ section below for details on installing the ``migration-assist`` tool.
+
+2. ``migration-assist postgres``
+
+   Creates the PostgreSQL database schema, and prepares it for Mattermost deployment by downloading the necessary migrations and applying them.
+
+3. ``migration-assist pgloader``
+
+   Generates a pgloader configuration from DSN values, ensuring accurate data transfer. See the :ref:`install pgloader <deploy/manual-postgres-migration:install pgloader>` documentation for details on installing the pgloader tool.
```

Comment:
Maybe we can move this section above previous place or remove it entirely if you think it's redundant, 0/5 on that. The fact is, this is not related to compiling the tool.

Reply:
Good point and thank you. I'd like to keep this information on the page because it provides technical context on the tool's components. Separated from the compile section.